src/Microcosm.js

```diff
@@ -39,68 +41,43 @@ let Microcosm = function() {
 Microcosm.prototype = {
   constructor: Microcosm,
 
-  /**
-   * Generates the initial state a microcosm starts with. Called whenever
-   * a microcosm runs start().
-   */
   getInitialState() {
-    return this.dispatch({}, lifecycle.willStart)
+    return this.dispatch({}, Transaction(lifecycle.willStart, this.state))
   },
 
-  /**
-   * Called before a transaction is squashed into base state. This method
-   * is useful to override if you wish to preserve transaction history
-   * beyond outstanding transactions.
-   */
-  shouldTransactionMerge(transaction) {
-    return transaction.meta.done
-  },
+  transactionWillUpdate(transaction, error, payload, complete) {
+    transaction.error    = error
+    transaction.payload  = payload
+    transaction.complete = complete
 
-  /*
-   * Before dispatching, this function is called on every transaction
-   */
-  shouldTransactionDispatch(transaction) {
-    return transaction.meta.active
-  },
-
-  /**
-   * Remove a transactions
-   */
-  release(transaction) {
-    this.transactions.splice(this.transactions.indexOf(transaction), 1)
-  },
-
-  transactionDidFail(transaction) {
-    this.release(transaction)
-    return this.rollforward()
+    if (!complete && has(this.transactions, transaction) === false) {
+      this.transactions = this.transactions.concat(transaction).sort(sortById)
+    }
   },
```

Comment:
Actually, I always want to push these in to maintain history for those who want it.